node/node.go

```diff
@@ -274,24 +277,49 @@ func (n *Node) run(ctx context.Context) (err error) {
 			select {
 			case <-agentDone:
 				return
-			case node := <-n.notifyNodeChange:
-				// If the server is sending us a ForceRenewal State, renew
-				if node.Certificate.Status.State == api.IssuanceStateRotate {
+			case nodeChanges := <-n.notifyNodeChange:
+				if nodeChanges.Node != nil {
+					// If the server is sending us a ForceRenewal State, renew
+					if nodeChanges.Node.Certificate.Status.State == api.IssuanceStateRotate {
+						renewCert()
+						continue
```

Comment:
`continue` doesn't seem right here, as that would ignore a change to the root certificate.

Splitting this code into a separate functions for handling node changes and root certificate changes may help make this clearer, but I fully admit I probably wouldn't bother with this myself.